transforms/__testfixtures__/class-test2.output.js

```diff
@@ -1,22 +1,78 @@
 'use strict';
 
 var React = require('React');
+var ReactComponentWithPureRenderMixin = require('ReactComponentWithPureRenderMixin');
+var FooBarMixin = require('FooBarMixin');
+
+class ComponentWithNonSimpleInitialState extends React.Component {
+  static iDontKnowWhyYouNeedThis = true; // but comment it
+  static foo = 'bar';
 
-// Comment
-module.exports = class extends React.Component {
   constructor(props, context) {
     super(props, context);
 
     this.state = {
-      foo: 'bar',
+      counter: props.initialNumber + 1,
     };
   }
 
   render() {
+    return (
+      <div>{this.state.counter}</div>
+    );
+  }
+}
+
+// Comment
+module.exports = class extends React.Component {
+  static propTypes = {
+    foo: React.PropTypes.bool,
+  };
+
+  static defaultProps = {
+    foo: 12,
+  };
+
+  state = function() { // non-simple
```

Comment:
This doesn’t look very good. I think most people will be confused by this, and will copy and paste this pattern even when it is unnecessary without understanding what it means.

Let’s embed non-trivial state initializers into the constructor.